DataFormats/Detectors/TPC/include/DataFormatsTPC/PIDResponse.h

```diff
@@ -97,12 +97,26 @@ GPUd() o2::track::PID::ID PIDResponse::getMostProbablePID(const TrackTPC& track,
   }
 
   auto id = o2::track::PID::Electron;
-  float distanceMin = o2::gpu::GPUCommonMath::Abs(dEdx - getExpectedSignal(track, id));
+  float distanceMin = 0.;
+  if (PID_useNsigma) {
+    // using nSigma
+    distanceMin = o2::gpu::GPUCommonMath::Abs((dEdx - getExpectedSignal(track, id)) / (PID_sigma * getExpectedSignal(track, id)));
+  } else {
+    // using absolute distance
+    distanceMin = o2::gpu::GPUCommonMath::Abs(dEdx - getExpectedSignal(track, id));
+  }
```

Comment:
I suggest to reduce the code a bit.
```suggestion
    const float dEdxExpected = getExpectedSignal(track, id);
    distanceMin = o2::gpu::GPUCommonMath::Abs(dEdx - dEdxExpected);
  if (PID_useNsigma) {
    // using nSigma
    distanceMin = distanceMin / (PID_sigma * dEdxExpected);
  } 
```
**Same below**